cmd/debug/debug.go

```diff
@@ -0,0 +1,215 @@
+			olderThanDuration := time.Duration(olderThan) * 24 * time.Hour
+
+			pruneChecksResults(db, olderThanDuration)
+		},
+	}
+
+	pruneCmd.Flags().UintVar(&olderThan, "older-than", 10, "Prune executed checks results data older than <value> days.")
+
+	debugCmd.AddCommand(pruneCmd)
+}
+
+func addDBResetCmd(debugCmd *cobra.Command) {
+	dbResetCmd := &cobra.Command{
+		Use:   "db-reset",
+		Short: "Reset the database",
+		Run: func(*cobra.Command, []string) {
+			db := initDB()
+
+			dbReset(db)
+		},
+	}
+
+	debugCmd.AddCommand(dbResetCmd)
+}
+
+func addDumpScenarioCmd(debugCmd *cobra.Command) {
+	dumpScenarioCmd := &cobra.Command{
+		Use:   "dump-scenario",
+		Short: "Dump the current scenario",
+		Run: func(*cobra.Command, []string) {
+			db := initDB()
+			exportPath := viper.GetString("path")
+			scenarioName := viper.GetString("name")
+
+			dumpScenario(db, exportPath, scenarioName)
+		},
+	}
+
+	var exportPath, scenarioName string
+
+	dumpScenarioCmd.Flags().StringVar(&exportPath, "path", ".", "The path where the scenario will be exported.")
+	dumpScenarioCmd.Flags().StringVar(&scenarioName, "name", "", "The scenario name.")
+	dumpScenarioCmd.MarkFlagRequired("name")
+
+	debugCmd.AddCommand(dumpScenarioCmd)
+}
+
+func initDB() *gorm.DB {
+	dbConfig := dbCmd.LoadConfig()
+	db, err := db.InitDB(dbConfig)
+	if err != nil {
+		log.Fatal("Error while initializing the database: ", err)
+	}
+
+	return db
+}
+
+func pruneEvents(db *gorm.DB, olderThan time.Duration) {
+	log.Infof("Pruning events older than %d days.", olderThan)
+
+	result := db.Delete(datapipeline.DataCollectedEvent{}, "created_at < ?", time.Now().Add(-olderThan))
+	log.Debugf("Pruned %d events", result.RowsAffected)
+
+	if result.Error != nil {
+		log.Fatalf("Error while pruning older events: %s", result.Error)
+	}
+	log.Infof("Events older than %d days pruned.", olderThan)
+}
+
+func pruneChecksResults(db *gorm.DB, olderThan time.Duration) {
+	log.Infof("Pruning checks results older than %d days.", olderThan)
+
+	result := db.Delete(entities.ChecksResult{}, "created_at < ?", time.Now().Add(-olderThan))
+	log.Debugf("Pruned %d checks results", result.RowsAffected)
+
+	if result.Error != nil {
+		log.Fatalf("Error while pruning older events: %s", result.Error)
+	}
+	log.Infof("Checks results older than %d days pruned.", olderThan)
+}
+
+func dbReset(db *gorm.DB) {
+	log.Info("Resetting database...")
+	err := db.Transaction(func(tx *gorm.DB) error {
+		for _, t := range web.DBTables {
+			stmt := &gorm.Statement{DB: db}
+			stmt.Parse(t)
+			tableName := stmt.Schema.Table
+
+			err := tx.Raw("TRUNCATE TABLE ?", tableName).Error
+			log.Infof("Table %s truncated.", tableName)
+			if err != nil {
+				log.Fatalf("Error while truncating table %s: %s", tableName, err)
+			}
+		}
+		return nil
+	})
+
+	if err != nil {
+		log.Fatal("Error while resetting the database: ", err)
+	}
+
+	log.Info("Database reset.")
+}
+
+func dumpScenario(db *gorm.DB, exportPath string, scenarioName string) {
```

Comment:
Nice. Would have any value to dump n number of entries? Instead of the latest one?
I would find this really useful for bug debugging.
Anyway, with the super big amount of records that we would have, it would still be difficult

Reply:
This actually export the latest events needed to build the latest state of the application, one per type per agent, by looking to the latest projected event from the subscriptions.